.github/workflows/_check-user.yml

```diff
@@ -0,0 +1,72 @@
+name: Check whether the workflow owner can use ARC runners
+
+on:
+  workflow_call:
+    inputs:
+      user_name:
+        required: true
+        type: string
+        description: The name of the workflow owner.
+      curr_branch:
+        required: true
+        type: string
+        description: Current branch.
+
+    outputs:
+      workflow-type:
+        description: Type of runners to use
+        value: ${{ jobs.check-user.outputs.workflow-type }}
+
+jobs:
+  check-user:
+    runs-on: linux.4xlarge
+    outputs:
+      workflow-type: ${{ steps.set-condition.outputs.workflow-type }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Check if user is in arc_users.yml
+        id: set-condition
+        run: |
+          curr_branch="${{ inputs.curr_branch }}"
+          echo "Current branch is '$curr_branch'"
+
+          if [[ "$curr_branch" == "nightly" || "$curr_branch" == "main" || "$curr_branch" == "release/"* || "$curr_branch" == "landchecks/"* ]]; then
+            echo "workflow-type=label" >> "$GITHUB_OUTPUT"
+          else
+            REPO_OWNER="pytorch"
+            REPO_NAME="test-infra"
+            ISSUE_NUMBER="5132"
+
+            COMMENTS=$(curl -H "Authorization: token $GITHUB_TOKEN" "https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/issues/$ISSUE_NUMBER/comments")
+            USER_LIST=$(echo "$COMMENTS" | tr -d '\r\n' | jq '.[0].body' -r)
+            RUN_OPTION=$(echo "$COMMENTS" | tr -d '\r\n' | jq '.[1].body' --raw-output)
+
+            WORKFLOW_TYPE_RG="rg"
+            WORKFLOW_TYPE_LABEL="label"
+            WORKFLOW_TYPE_BOTH="both"
+
+            if [[ ${USER_LIST:0:1} == "!" ]]; then
```

Comment:
How do you feel about turning the comment into properly formatted yaml or json?

That way you can cleanly put everything in one block without having to worry as much about parsing special symbols

It'll also let us gradually shift from "always running in both locations" to "start opting people into ARC-only mode"

Eg. the comment block could look like:

```
Modes:
  - type: both
     users: # list of users opted into both
        - atalman
        - clee2000
        - ...
  - type: arc-only
     users: # users opted into  ARC-only
     - ZainRizvi
     - DanilBaibak
  - type: classic-only
     users:
     - * # default result

```

Reply:
My main idea is to make the solution simple. With a single list of people it is quite straightforward:
- If you are in the list, you use ARC, if not, then you don’t.
- You can also turn it on for everyone or turn it off.

There is no goal to maintain this system forever. The goal is to stabilize the ARC runners and only use them.